content/TypeHinting/TypeHinting.q.md

```diff
@@ -0,0 +1,1346 @@
+# Type Hinting
+
++++
+
+### Part 1: Sample Code Analysis
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+declare(strict_types = 1);
+
+class Person 
+{
+  protected $name;
+
+  public function setName(string $name)
+  {
+    $this->name = $name;
+  }
+
+  public function getName() : string
+  {
+    return $this->name;
+  }
+}
+
+$personObject = new Person();
+$personObject->setName("Anna");
+echo "Person name: " . $personObject->getName();
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+ - It prints `Anna`.
+
+ - It prints `$name`.
+
+ - It prints `Person`.
+
+ - It prints `$personObject`.
+
+ - It prints `Person name: Anna`.
+
+
+/// type=SS, answer=[3]
+
+Which of the following is a class?
+
+ - `Anna`
+
+ - `$name`
+
+ - `Person`
+
+ - `getName()`
+
+ - `$personObject`
+
+
+/// type=SS, answer=[2]
+
+Which of the following is a property?
+
+ - `Anna`
+
+ - `$name`
+
+ - `Person`
+
+ - `getName()`
+
+ - `$personObject`
+
+
+/// type=MS, answer=[3,4]
+
+Which of the following are methods?
+
+ - `$name`
+
+ - `Person`
+
+ - `setName()`
+
+ - `getName()`
+
+ - `$personObject`
+
+
+/// type=SS, answer=[4]
+
+In the statement `declare(strict_types = 1);` on line 2, what is `declare`?
+
+ - It is a keyword.
+
+ - It is a directive.
+
+ - It is a data type.
+
+ - It is a construct.
+
+ - It is an operator.
+
+
+/// type=SS, answer=[2]
+
+In the statement `declare(strict_types = 1);` on line 2, what is `strict_types`?
+
+ - It is a keyword.
+
+ - It is a directive.
+
+ - It is a data type.
+
+ - It is a construct.
+
+ - It is an operator.
+
+
+/// type=SS, answer=[2]
+
+On line 2, what does `declare(strict_types = 1);` do?
+
+ - It sets the value `1` as a default type declaration.
+
+ - It enables strict type mode validations within the file.
+
+ - It creates the `declare()` method of the `Person` class.
```

Comment:
I think there is no problem with this one since this is just a distractor and not given as the correct answer. This is to check if students really understands or examine the given code. Thank you.

Reply:
Noted, thank you.